src/schema/formats.js

```diff
@@ -103,6 +103,37 @@ export function isSecureUrl(value) {
   return ['https:', 'wss:'].includes(url.protocol);
 }
 
+export function isOrigin(value) {
+  // FIXME: should we only have isSecureOrigin ?
+  let url;
+  try {
+    url = new URL(value);
+  } catch (e) {
+    // It's invalid or not absolute.
+    return false;
+  }
+  if (value.includes('*')) {
+    // Wildcards are not valid in origins.
+    return false;
+  }
+  if (
+    url.pathname !== '/' ||
+    url.hash !== '' ||
+    url.search !== '' ||
+    value.endsWith('/')
+  ) {
+    // Path, query string and hash shouldn't be included in origins.
+    // URL().pathname will always be '/' if no path is provided, so we have to
+    // check value doesn't end with one either.
+    return false;
+  }
+  // value === url.origin would be enough if it wasn't for IDNs but
+  //  URL().origin returns punycode, so we have to compare against URL().href
+  // minus the last character instead, having checked that there was no path,
+  // query string or hash earlier.
+  return url.href.slice(0, -1) === url.origin;
```

Comment:
We should keep this as much as possible in sync with what we are going to do on the Firefox side, when the patch on the Firefox side will be landed it may be a good idea to put a permalink to the formatter used in Firefox.

In the meantime, here is the related review comment I added on the Firefox patch:
https://phabricator.services.mozilla.com/D129105#inline-715951

Shane and I haven't touched base about that yet, but on the Firefox side we will proceed to use a Schemas formatter like the one I described in that phabricator comment, then in nodejs I think that it may look more or less like the following code snippet:
```js
const ALLOWED_INSTALL_ORIGINS_PROTOCOLS = ["http:", "https:"];

...

function isValidInstallOriginUrl(value) {
  let origin;
  try {
    const url = new URL(value);
    if (!ALLOWED_INSTALL_ORIGINS_PROTOCOLS.includes(url.protocol)) {
      // URL protocol is not one of the allowed ones (e.g. ["http:", "https:"]).
      return false;
    }
    origin = url.origin;
  } catch (err) {
    // Not a valid url.
    return false;
  }

  const { href: baseURL } = new URL("/", origin);
  // Make sure value URL doesn't include any other URL component besides the origin
  // and path set to "/". 
  return baseURL.slice(0, -1) === value;
} 
```